server/vulnerabilities/sync.go

```diff
@@ -0,0 +1,262 @@
+package vulnerabilities
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/facebookincubator/nvdtools/cvefeed"
+	feednvd "github.com/facebookincubator/nvdtools/cvefeed/nvd"
+	"github.com/fleetdm/fleet/v4/pkg/download"
+	"github.com/fleetdm/fleet/v4/pkg/fleethttp"
+	"github.com/fleetdm/fleet/v4/server/config"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/ptr"
+	"github.com/hashicorp/go-multierror"
+	"go.uber.org/multierr"
+)
+
+func Sync(vulnPath string, config config.FleetConfig, ds fleet.Datastore) error {
+	if config.Vulnerabilities.DisableDataSync {
+		return nil
+	}
+
+	client := fleethttp.NewClient()
+
+	var syncErr error
+
+	if err := DownloadCPEDatabase(vulnPath, client, WithCPEURL(config.Vulnerabilities.CPEDatabaseURL)); err != nil {
+		syncErr = multierror.Append(syncErr, fmt.Errorf("sync CPE database: %w", err))
+	}
+
+	if err := DownloadNVDCVEFeed(vulnPath, ""); err != nil {
+		syncErr = multierr.Append(syncErr, fmt.Errorf("sync NVD CVE feed: %w", err))
+	}
+
+	if err := DownloadEPSSFeed(vulnPath, client); err != nil {
+		syncErr = multierr.Append(syncErr, fmt.Errorf("sync EPSS CVE feed: %w", err))
+	}
+
+	if err := DownloadCISAKnownExploitsFeed(vulnPath, client); err != nil {
+		syncErr = multierr.Append(syncErr, fmt.Errorf("sync CISA known exploits feed: %w", err))
+	}
+
+	if err := LoadCVEScores(vulnPath, ds); err != nil {
+		syncErr = multierr.Append(syncErr, err)
+	}
+
+	return syncErr
+}
+
+const epssFeedsURL = "https://epss.cyentia.com"
+const epssFilename = "epss_scores-current.csv.gz"
+
+func DownloadEPSSFeed(vulnPath string, client *http.Client) error {
+	urlString := epssFeedsURL + "/" + epssFilename
+	u, err := url.Parse(urlString)
+	if err != nil {
+		return fmt.Errorf("parse url: %w", err)
+	}
+	path := filepath.Join(vulnPath, strings.TrimSuffix(epssFilename, ".gz"))
+
+	err = download.Download(client, u, path)
+	if err != nil {
+		return fmt.Errorf("download %s: %w", u, err)
+	}
+
+	return nil
+}
+
+// EPSSScore represents the EPSS score for a CVE.
+type EPSSScore struct {
```

Comment:
Seems like this type is only used in this module, do we want to make this public?